dpnp/backend/kernels/dpnp_krnl_statistics.cpp

```diff
@@ -1,200 +0,0 @@
-// Copyright (c) 2016-2024, Intel Corporation
-// All rights reserved.
-//
-// Redistribution and use in source and binary forms, with or without
-// modification, are permitted provided that the following conditions are met:
-// - Redistributions of source code must retain the above copyright notice,
-//   this list of conditions and the following disclaimer.
-// - Redistributions in binary form must reproduce the above copyright notice,
-//   this list of conditions and the following disclaimer in the documentation
-//   and/or other materials provided with the distribution.
-//
-// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
-// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
-// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
-// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
-// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
-// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
-// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
-// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
-// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
-// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
-// THE POSSIBILITY OF SUCH DAMAGE.
-//*****************************************************************************
-
-#include <iostream>
-
-#include "dpnp_fptr.hpp"
-#include "dpnp_utils.hpp"
-#include "dpnpc_memory_adapter.hpp"
-#include "queue_sycl.hpp"
-#include <dpnp_iface.hpp>
-
-namespace mkl_blas = oneapi::mkl::blas::row_major;
-namespace mkl_stats = oneapi::mkl::stats;
-
-template <typename _KernelNameSpecialization1,
-          typename _KernelNameSpecialization2,
-          typename _KernelNameSpecialization3>
-class dpnp_correlate_c_kernel;
-
-template <typename _DataType_output,
-          typename _DataType_input1,
-          typename _DataType_input2>
-DPCTLSyclEventRef dpnp_correlate_c(DPCTLSyclQueueRef q_ref,
-                                   void *result_out,
-                                   const void *input1_in,
-                                   const size_t input1_size,
-                                   const shape_elem_type *input1_shape,
-                                   const size_t input1_shape_ndim,
-                                   const void *input2_in,
-                                   const size_t input2_size,
-                                   const shape_elem_type *input2_shape,
-                                   const size_t input2_shape_ndim,
-                                   const size_t *where,
-                                   const DPCTLEventVectorRef dep_event_vec_ref)
-{
-    (void)where;
-
-    shape_elem_type dummy[] = {1};
-    return dpnp_dot_c<_DataType_output, _DataType_input1, _DataType_input2>(
```

Comment:
I think you can remove legacy implementation of `dot` as well since it was only used here in `correlate`. In addition, I believe legacy implementation of `multiply` can also be removed, since it is only used in `dot`.

Reply:
thanks for pointing this out!
I've created a separate pull-request for this in order not to overburden this one.
https://github.com/IntelPython/dpnp/pull/2183

I would also move deletion of backend implementation of correlate to the new PR.